sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/DataFeedLiveTests.cs

```diff
@@ -0,0 +1,897 @@
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
+
+            var granularity = new DataFeedGranularity(DataFeedGranularityType.Daily);
+            var schema = new DataFeedSchema(metrics);
+            var ingestionSettings = new DataFeedIngestionSettings(ingestionStartTime);
+
+            return new DataFeed(name, dataSource, granularity, schema, ingestionSettings);
+        }
+
+        private DataFeed GetDataFeedWithOptionalMembersSet(string name, DataFeedSource dataSource)
+        {
+            var metrics = new List<DataFeedMetric>()
+            {
+                new ("cost") { MetricDisplayName = "costDisplayName", MetricDescription = "costDescription" },
+                new ("revenue") { MetricDisplayName = "revenueDisplayName", MetricDescription = "revenueDescription" }
+            };
+            var dimensionColumns = new List<DataFeedDimension>()
+            {
+                new ("city"),
+                new ("category") { DimensionDisplayName = "categoryDisplayName" }
+            };
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
+
+            var granularity = new DataFeedGranularity(DataFeedGranularityType.Custom) { CustomGranularityValue = 1360 };
+            var schema = new DataFeedSchema(metrics) { DimensionColumns = dimensionColumns, TimestampColumn = "timestamp" };
+            var ingestionSettings = new DataFeedIngestionSettings(ingestionStartTime)
+            {
+                IngestionStartOffset = TimeSpan.FromMinutes(30),
+                IngestionRetryDelay = TimeSpan.FromSeconds(80),
+                StopRetryAfter = TimeSpan.FromMinutes(10),
+                DataSourceRequestConcurrency = 5
+            };
+
+            return new DataFeed(name, dataSource, granularity, schema, ingestionSettings)
+            {
+                Description = "This data feed was created to test the .NET client.",
+                AccessMode = DataFeedAccessMode.Public,
+                ActionLinkTemplate = "https://fakeurl.com/%metric/%datafeed",
+                Administrators = new List<string>() { "fake@admin.com" },
+                Viewers = new List<string>() { "fake@viewer.com" },
+                MissingDataPointFillSettings = new () { FillType = DataFeedMissingDataPointFillType.CustomValue, CustomFillValue = 45.0 }
+            };
+        }
+
+        private void ValidateDataFeedWithMinimumSetup(DataFeed dataFeed, string expectedId, string expectedName)
+        {
+            var ingestionStartTime = DateTimeOffset.Parse("2020-08-01T00:00:00Z");
+
+            Assert.That(dataFeed.Id, Is.EqualTo(expectedId));
+            Assert.That(dataFeed.Name, Is.EqualTo(expectedName));
+            Assert.That(dataFeed.Description, Is.Not.Null.And.Empty);
+            Assert.That(dataFeed.Status, Is.EqualTo(DataFeedStatus.Active));
+            Assert.That(dataFeed.AccessMode, Is.EqualTo(DataFeedAccessMode.Private));
```

Comment:
is this the default of the service? Should it actually be nullable on our side?

Reply:
We use the same model for the `UpdateDataFeed` operation, and an `AccessMode` doesn't need to be specified. So we're making it nullable.